libevmasm/Assembly.cpp

```diff
@@ -53,6 +53,54 @@ using namespace solidity::evmasm;
 using namespace solidity::langutil;
 using namespace solidity::util;
 
+namespace
+{
+
+/// Produces instruction location info in RAII style. When an assembly instruction is added to the bytecode,
+/// this class can be instantiated in that scope. It will record the current bytecode size (before addition)
+/// and, at destruction time, record the new bytecode size. This information is then added to an external
+/// instruction locations vector.
+/// If the instruction decomposes into multiple individual evm instructions, `emit` can be
+/// called for all but the last one (which will be emitted by the destructor).
+class AddInstructionLocation
```

Comment:
How about this? The way it's now, it sounds more like a function.

```suggestion
class InstructionLocationEmitter
```

Reply:
Alternatively:
```suggestion
class InstructionLocationStamper
```
with `emit()` being renamed to `stamp()`.

Reply:
I like the emitter one. :) Refactored the name.